src/xunit.performance.api/ScenarioConfiguration.cs

```diff
@@ -13,28 +14,23 @@ public sealed class ScenarioConfiguration
         /// Initializes a new instance of the ScenarioConfiguration class.
         /// </summary>
         /// <param name="timeSpan">The amount of time to wait for one iteration process to exit.</param>
-        public ScenarioConfiguration(TimeSpan timeSpan)
+        /// <param name="startInfo">The data with which to start the scenario.</param>
+        public ScenarioConfiguration(TimeSpan timeSpan, ProcessStartInfo startInfo)
         {
             if (timeSpan.TotalMilliseconds < 1)
                 throw new InvalidOperationException("The time out per iteration must be a positive number.");
 
+            Debug.Assert(
+                !(startInfo.UseShellExecute && (startInfo.RedirectStandardError || startInfo.RedirectStandardInput || startInfo.RedirectStandardOutput)),
```

Comment:
Do you need this assert?  I'd expect the Process class to handle this.

Reply:
I'll remove it. I was using it for debugging/testing my scenarios. It is a left over.

---
In reply to: [143035933](https://github.com/Microsoft/xunit-performance/pull/241#discussion_r143035933) [](ancestors = 143035933)